src/app/store.ts

```diff
@@ -11,14 +11,47 @@ import {
   UrlInputState,
 } from "../features/urlInput";
 import { TabsActions, tabsReducer, TabsState } from "../features/tabs";
-
 export type ApplicationState = {
   knownTags: KnownTagsState;
   items: ItemsState;
   urlInput: UrlInputState;
   tabs: TabsState;
 };
-export default configureStore<
+
+const localStorages = ["knownTags", "items", "tabs"];
+
+function saveToLocalStorage(state: ApplicationState) {
+  try {
+    localStorages.map((storage) => {
+      const storePart = storage as keyof typeof state;
+      const serialisedState = JSON.stringify(state[storePart]);
+      localStorage.setItem("LinkBiscuit_" + storage, serialisedState);
+    });
+  } catch (e) {
+    console.warn(e);
+  }
+}
+
+function loadFromLocalStorage() {
+  try {
+    let loadedState = "";
+    localStorages.map((storage) => {
+      const unserializedState = localStorage.getItem("LinkBiscuit_" + storage);
+      if (unserializedState !== null) {
+        loadedState += '"' + storage + '":' + unserializedState + ",";
+      }
+    });
+    return loadedState
+      ? JSON.parse("{" + loadedState.replace(/,$/, "") + "}")
```

Comment:
That's a bit strange. Why is not `JSON.parse(...)` enough?

Reply:
Yeah, it is strange, you are right.